app/conode/exit_conodes.sh

```diff
@@ -1,6 +1,10 @@
 #!/bin/bash
 
-HOSTS=$( cat config.toml | grep Hosts | sed -e "s/.*\[\"\(.*\)\"\]/\1/" | perl -pe "s/\", \"/\n/g" )
+CONFIG=config.toml
+if [ "$1" ]; then
+  CONFIG=$1/$CONFIG
+fi
+HOSTS=$( grep Hosts $CONFIG | sed -e "s/.*\[\"\(.*\)\"\]/\1/" | perl -pe "s/\", \"/\n/g" | tail -r )
```

Comment:
Le 16/12/2015 14:21, Ismail a écrit :
>
> In app/conode/exit_conodes.sh 
> <https://github.com/dedis/cothority/pull/142#discussion_r47774281>:
>
> > @@ -1,6 +1,10 @@
> >  #!/bin/bash
> >
> > -HOSTS=$( cat config.toml | grep Hosts | sed -e "s/.*\[\"\(.*\)\"\]/\1/" | perl -pe "s/\", \"/\n/g" )
> > +CONFIG=config.toml
> > +if [ "$1" ]; then
> > +  CONFIG=$1/$CONFIG
> > +fi
> > +HOSTS=$( grep Hosts $CONFIG | sed -e "s/.*\[\"\(.*\)\"\]/\1/" | perl -pe "s/\", \"/\n/g" | tail -r )
>
> Note: after merging we've recognized that |tail -r| won't work on most 
> linux machines...
>
>
There is `tac` - but I think that doesn't exist everywhere, neither. At 
least it doesn't on a mac. I wanted to exit the leaf-nodes as first 
nodes, then the root-node. But in fact it doesn't do it hierarchly 
anyway. So you might as well just throw away the `tail -r`.